### PR TITLE
DRY radian formatting

### PR DESCRIPTION
**Pull request by viathor**
Replace private functions `_rads_func_symbol` and `_format_rads` with a new public method `CircuitDiagramInfoArgs.format_radians` and re-use wherever appropriate.

### CODE REVIEW

File: cirq/ops/common_gates.py [hunk 1]
```diff
@@ -492,10 +492,10 @@ def _has_stabilizer_effect_(self) -> Optional[bool]:
...
                               ) -> Union[str, 'protocols.CircuitDiagramInfo']:
         if self._global_shift == -0.5:
-            return _rads_func_symbol(
-                'Rz',
-                args,
-                self._diagram_exponent(args, ignore_global_phase=False))
+            exponent = self._diagram_exponent(args, ignore_global_phase=False)
+            pi = sympy.pi if protocols.is_parameterized(exponent) else np.pi
+            angle_str = args.format_radians(radians=exponent * pi)
+            return f'Rz({angle_str})'
```
Comment (dstrain115) [inline cirq/ops/common_gates.py:498]: We have this code copied three times in this file.  Is it worth writing a small helper function?

Comment (viathor) [reply cirq/ops/common_gates.py:492]: Done.

File: cirq/ops/parity_gates.py [hunk 2]
```diff
@@ -88,9 +89,11 @@ def _circuit_diagram_info_(self, args: 'cirq.CircuitDiagramInfoArgs'
...
         if self._global_shift == -0.5:
             # Mølmer–Sørensen gate.
-            symbol = common_gates._rads_func_symbol(
-                'MS', args,
-                self._diagram_exponent(args, ignore_global_phase=False) / 2)
+            exponent = self._diagram_exponent(args,
+                                              ignore_global_phase=False) / 2
+            pi = sympy.pi if protocols.is_parameterized(exponent) else np.pi
+            angle_str = args.format_radians(radians=exponent * pi)
+            symbol = f'MS({angle_str})'
```
Comment (dstrain115) [inline cirq/ops/parity_gates.py:96]: This is an eigengate, can it use _format_exponent_as_angle()?

Comment (viathor) [reply cirq/ops/parity_gates.py:93]: Done.

**Pull request merged**